benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/goal/explorationspace/workload/users/UserValuesYamlProtocol.scala

```diff
@@ -0,0 +1,162 @@
+package cloud.benchflow.dsl.definition.configuration.goal.explorationspace.workload.users
+
+import cloud.benchflow.dsl.definition.configuration.goal.explorationspace.ExplorationSpaceYamlProtocol
+import cloud.benchflow.dsl.definition.configuration.goal.explorationspace.step.RangeWithStep
+import cloud.benchflow.dsl.definition.errorhandling.BenchFlowDeserializationException
+import cloud.benchflow.dsl.definition.errorhandling.YamlErrorHandler.{ deserializationHandler, unsupportedReadOperation, unsupportedWriteOperation }
+import net.jcazevedo.moultingyaml.{ DefaultYamlProtocol, YamlFormat, YamlObject, YamlString, YamlValue, _ }
+
+import scala.util.{ Failure, Success, Try }
+
+/**
+ * @author Jesper Findahl (jesper.findahl@gmail.com)
+ *         created on 2017-05-23
+ */
+object UserValuesYamlProtocol extends DefaultYamlProtocol {
```

Comment:
You deleted https://github.com/benchflow/benchflow/pull/465/files#diff-5b7b9c50538d06a01fd7c3dca9fa2dd4 and added this class. This class has much code that is similar to other classes (or it is going to be similar).

According to me we should:

- Have base types (equivalent to primitive types in a programming language), that are defined independently of the DSL entity so that can be reused to obtain values. In this case `Int` would be an independent type, and `User` would use it. This way you can use the same definition to get the value of `users` independently of where in the DSL it is defined (since we allow it to be defined as exploration parameter or as configuration. 
- Have a generic definition for shared elements, as `values`, `range`, `step` that can be reused for all the elements needing it. This generic definition is reused, and according to the context (e.g., reading the `users` or the `memory` definition) the correct Reader can be called. 


Update this PR or open an issue to describe the needed changes.